arch/risc-v/src/bl602/bl602_systemreset.c

```diff
@@ -47,19 +53,33 @@
  *
  ****************************************************************************/
 
-static void bl602_chip_reset(uint32_t mask)
+void __attribute__((section(".tcm_code"))) bl602_chip_reset(uint32_t mask)
 {
+  /* We choose to use ROM driver */
+
+#if 0
+  uint32_t regval;
+
+  /* Disable interrupt */
+
+  asm volatile("csrci mstatus, 8");
+
   /* Reset the root clock */
 
-  modifyreg32(BL602_HBN_GLB, HBN_GLB_HBN_ROOT_CLK_SEL_MASK, 0);
```

Comment:
Why did you remove modifyreg32. This is the atomic operation?

Reply:
Because modifyreg32 is the code on XIP Flash, and no code on XIP Flash can be called during reset.